worker-delivery-validate.py

```
@@ -0,0 +1,127 @@
+    if pr_head != head: fail("PR head %s does not match tested checkout head %s" % (pr_head[:12], head[:12]))
+    live = {}
+    def keep(key, when, bucket, source):
+        cur = live.get(key)
+        if not cur or when >= cur["when"]: live[key] = {"bucket": bucket, "source": source, "when": when}
+    def when(value):
+        if not value: return datetime.min
+        try: return datetime.fromisoformat(value.replace("Z", "+00:00"))
+        except ValueError: return datetime.min
+    for item in gh_api(gh, "repos/%s/commits/%s/check-runs" % (repo, pr_head)).get("check_runs") or []:
+        name = str(item.get("name", "")).strip()
+        if not name: continue
```

Ruff (0.16.3)

[error] Multiple statements on one line (colon)

(E701)